lib/vagrant-service-manager/command.rb

```diff
@@ -172,6 +175,31 @@ def find_machine_ip
         end
       end
 
+      def sha_id(file_data)
+        Digest::SHA256.hexdigest file_data
+      end
+
+      def certs_present_and_valid?(path, machine)
+        return false unless File.directory?(path)
+
+        # check validity of certs
+        Dir[path + "/*"].each do |f|
+          name = File.basename(f)
+
+          guest_file_path = "#{DOCKER_PATH}/#{name}"
+          machine.communicate.execute("cat #{guest_file_path}") do |type, data|
+            if type == :stderr
+              @ui.error(data)
+              exit 126
+            end
+
+            return false if sha_id(File.read(f)) != sha_id(data)
```

Comment:
Right now you are downloading the cert from the VM and calculating the sha on in ruby.  Would it be more efficient to run the sha on the VM and just get that info?